examples/chat2-reliable/chat.go

```diff
@@ -137,20 +151,65 @@ func (c *Chat) receiveMessages() {
 		case <-c.ctx.Done():
 			return
 		case value := <-c.C:
-
 			msgContentTopic := value.Message().ContentTopic
 			if msgContentTopic != c.options.ContentTopic {
 				continue // Discard messages from other topics
 			}
 
 			msg, err := decodeMessage(c.options.ContentTopic, value.Message())
 			if err == nil {
-				// send valid messages to the UI
-				c.ui.ChatMessage(int64(msg.Timestamp), msg.Nick, string(msg.Payload))
+				c.processReceivedMessage(msg)
 			}
 		}
 	}
 }
+
+func (c *Chat) processReceivedMessage(msg *pb.Message) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	// Update Lamport timestamp
+	if msg.LamportTimestamp > c.lamportTimestamp {
+		c.lamportTimestamp = msg.LamportTimestamp
+	}
+	c.lamportTimestamp++
```

Comment:
I need to think about this. Not sure it's necessary/desirable to increase the `lamportTimestamp` after every received message (certainly necessary before sending a new message). However, I can't currently think of a good reason _not_ to do this. 🤔